src/channels/media.rs

```diff
@@ -60,6 +60,144 @@ impl ToString for StreamType {
     }
 }
 
+/// Generic, movie, TV show, music track, or photo metadata.
+#[derive(Debug)]
+pub enum Metadata {
+    Generic(GenericMediaMetadata),
+    Movie(MovieMediaMetadata),
+    TvShow(TvShowMediaMetadata),
+    MusicTrack(MusicTrackMediaMetadata),
+    Photo(PhotoMediaMetadata),
+}
+
+/// Generic media metadata.
+///
+/// See also https://developers.google.com/cast/docs/reference/messages#GenericMediaMetadata.
```

Comment:
__nit:__ could you please convert these links (here and below) into proper markdown links (suggested by `clippy`)?

Reply:
Done.